hca/util/__init__.py

```diff
@@ -302,6 +301,17 @@ def login(self, access_token=""):
                                         token_type="Bearer")
         print("Storing access credentials")
 
+    def _atomic_write_to_file(self, filename, dir, content):
+        """
+        Atomically writes content a to file.
+
+        Creates a temp file and renames (atomic) file to desired filename.
+        To ensure atomicity, the src and dest dirs must be the same.
+        """
+        with tempfile.NamedTemporaryFile('wb', dir=dir, delete=False) as tmp:
+            tmp.write(content)
+            os.rename(tmp.name, filename)
```

Comment:
I believe that on Windows, it's not kosher to rename a file that's open.  Consider creating the tempfile with `delete=False`, close it, and then renaming it.  Then wrap the entire try-finally with something that cleans up the tempfile in case the rename fails.

Reply:
Also note from https://docs.python.org/2/library/os.html#os.rename:

> On Windows, if dst already exists, OSError will be raised even if it is a file

Here's some possible strategies for defeating that: https://bugs.python.org/issue8828